src/v/kafka/server/handlers/details/isolated_node_handler.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2021 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "cluster/metadata_cache.h"
+#include "kafka/server/request_context.h"
+#include "seastarx.h"
+
+namespace kafka {
+
+using is_node_isolated_or_decomissioned
+  = ss::bool_class<struct is_node_isolated_or_decomissioned_tag>;
+
+inline is_node_isolated_or_decomissioned
+node_isolated_or_decomissioned(request_context& ctx) {
```

Comment:
nit: maybe name this file `isolated_node_utils` ? handler in a name suggest that this is actually a request handler.